src/Microsoft.AspNetCore.Razor/Editor/ImplicitExpressionEditHandler.cs

```diff
@@ -149,10 +154,65 @@ private static bool IsSecondaryDotlessCommitInsertion(Span target, TextChange ch
                    change.OldLength == 0;
         }
 
-        private static bool IsAcceptableReplace(Span target, TextChange change)
+        private bool IsAcceptableReplace(Span target, TextChange change)
         {
             return IsEndReplace(target, change) ||
-                   (change.IsReplace && RemainingIsWhitespace(target, change));
+                (change.IsReplace && RemainingIsWhitespace(target, change));
+        }
+
+        private bool IsAcceptableIdentifierReplacement(Span target, TextChange change)
+        {
+            if (!change.IsReplace)
+            {
+                return false;
+            }
+
+            for (var i = 0; i < target.Symbols.Count; i++)
+            {
+                var symbol = target.Symbols[i] as CSharpSymbol;
+
+                if (symbol == null)
+                {
+                    break;
+                }
+
+                var symbolStartIndex = target.Start.AbsoluteIndex + symbol.Start.AbsoluteIndex;
+                var symbolEndIndex = symbolStartIndex + symbol.Content.Length;
```

Comment:
There is no Length property. We need to inspect the Content and then look at the length for symbols. You're probably thinking of Spans 😄